core/src/main/java/org/elasticsearch/indices/recovery/StartRecoveryRequest.java

```diff
@@ -43,7 +44,9 @@
 
     private Store.MetadataSnapshot metadataSnapshot;
 
-    private RecoveryState.Type recoveryType;
+    private org.elasticsearch.cluster.routing.RecoverySource recoverySource;
```

Comment:
yes, I was wondering if I should remove it. StartRecoveryRequest is always a peer recovery (from primary). We could also reflect that in the name of `StartRecoveryRequest`.